Library/Homebrew/utils/ruby_check_version_script.rb

```diff
@@ -18,6 +18,6 @@
   homebrew_required_ruby_version.canonical_segments
 
 if ruby_version_major != homebrew_required_ruby_version_major ||
```

Comment:
I wonder if we should allow major version deviation too; Ruby 3.0 isn't actually that incompatible with Ruby 2.6 (compared to Ruby 2.5).

Relatedly, may be worth allowing `HOMEBREW_DEVELOPER` to use higher versions but not lower ones. This is best done with a `Gem::Version >= Gem::Version` comparison.

Reply:
I would be surprised if the kwarg changes in Ruby 3.0 didn't cause any issues. Some of those issues may also be subtle.

Reply:
@Bo98 🤷🏻 I feel like that's why this is an undocumented, developer-only thing.

Reply:
Although this does make me think it's probably better to adding a undocumented `HOMEBREW_ALLOW_ANY_RUBY_VERSION` that is required as well as `HOMEBREW_DEVELOPER` to avoid screwing with people and CI where `HOMEBREW_DEVELOPER` is set.

Reply:
So... Yay or nay on the current version?